src/gpio_wrapper.cpp

```diff
@@ -39,17 +40,7 @@ void set_gpio_pin_output(uint32_t pin_number)
 
 void set_gpio_pin_input(uint32_t pin_number, InputResistor input_resistor)
 {
-    switch (input_resistor) {
-        case InputResistor::NONE:
-            nrf_gpio_cfg_input(pin_number, NRF_GPIO_PIN_NOPULL);
-            break;
-        case InputResistor::PUR:
-            nrf_gpio_cfg_input(pin_number, NRF_GPIO_PIN_PULLUP);
-            break;
-        case InputResistor::PDR:
-            nrf_gpio_cfg_input(pin_number, NRF_GPIO_PIN_PULLDOWN);
-            break;
-    }
+    nrf_gpio_cfg_input(pin_number, static_cast<nrf_gpio_pin_pull_t>(input_resistor));
```

Comment:
i think switching is more readable - dont really want to have to look at the sdk and enum match

Reply:
eh ig u put the enum in a way that you don't have to, I still think its more readable to switch. dont see much edge in the static cast

Reply:
its not to make it more readable, it was getting hard and very unreadable to have multiple switches since we had multiple diff enums

in general i dislike wrappers that dont abstract, but i can see this is subjective